pkg/sfu/streamallocator.go

```diff
@@ -173,11 +174,13 @@ func (s *StreamAllocator) Start() {
 }
 
 func (s *StreamAllocator) Stop() {
-	s.chMu.Lock()
-	defer s.chMu.Unlock()
+	if !s.isStopped.TrySet(true) {
+		return
+	}
 
-	close(s.runningCh)
+	s.eventChMu.Lock()
```

Comment:
should this lock around the TrySet block too? to ensure `isStopped` cannot be set to true *after* `if s.isStopped.Get()` in L308

Reply:
Good point @davidzhao . I think I need to look at few more place where we are using this pattern. Will scan for them today.